configure

```diff
@@ -604,10 +600,10 @@ def configure_node(o):
   o['variables']['host_arch'] = host_arch
   o['variables']['target_arch'] = target_arch
 
-  if target_arch != host_arch and options.with_snapshot:
-    o['variables']['want_separate_host_toolset'] = 1
-  else:
+  if target_arch != host_arch and options.without_snapshot:
     o['variables']['want_separate_host_toolset'] = 0
+  else:
+    o['variables']['want_separate_host_toolset'] = 1
```

Comment:
I suspect that the logic is wrong here because it will set `want_separate_host_toolset = 1` when `target_arch == host_arch`.